lib/sorcery/providers/linkedin.rb

```diff
@@ -45,6 +46,15 @@ def process_callback(params, _session)
 
         get_access_token(args, token_url: token_url, token_method: :post)
       end
+
+      def get_user_info(access_token)
+        response       = access_token.get(user_info_url)
+        email_response = access_token.get(email_info_url)
```

Comment:
I'm concerned about this breaking when an application isn't trying to use the `r_emailaddress` scope. @JakubSzajna can you test this while passing only `r_liteprofile` to ensure it still works?

Reply:
You are right @athix 
This breaks the application as there is an `OAuth2::Error` exception.
I've forced `r_emailaddress` to exist in `@scope` as skipping the email request would get us back to the beginning.
I've tested it and it worked in my app.

Reply:
Forcing a scope is not ideal, imo. Instead, I think we should check the scope for `r_emailaddress` and conditionally do the email_response call / merge.

```ruby
def scope_includes_r_emailaddress?
  scope = @scope.split(' ')
  scope.include?('r_emailaddress')
end
```

```ruby
if scope_includes_r_emailaddress?
  email_info = JSON.parse([...])

  user_info.merge([...])
end

user_info
```

Reply:
Ah, I've missed your solution and did it not exactly the same way, but it works 👍